### PR TITLE
[FLINK-36148][pipeline-connector][mysql] Add custom parser for CreateTableEvent

### PR DESCRIPTION
**Pull request by lvyanquan**
Support parsing table creation ddl and creating CreateTableEvent in Binlog reading phase.

### CODE REVIEW

**Comment by lvyanquan** [comment]
A follow up of https://github.com/apache/flink-cdc/pull/3560.
@leonardBang  PTAL.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlPipelineITCase.java [hunk 1]
```diff
@@ -501,6 +501,22 @@ public void testSchemaChangeEvents() throws Exception {
             expected.add(
                     new DropTableEvent(
                             TableId.tableId(inventoryDatabase.getDatabaseName(), "customers")));
+
+            // Test create table DDL
+            statement.execute(
+                    String.format(
+                            "CREATE TABLE `%s`.`newlyAddedTable1`(id int, id2 int, primary key(id));",
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlPipelineITCase.java:508]: Could you cover more data types like our full types test?

**Review by leonardBang — COMMENTED** [review]
Thanks @lvyanquan for the fix, I only left one minor comment about test.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlPipelineITCase.java (hunk 1 above)
Comment (lvyanquan) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlPipelineITCase.java]: Done.

**Review by leonardBang — APPROVED** [review]
Thanks for the update, LGTM

**Pull request merged**